lib/Tool/Text.php

```diff
@@ -83,7 +83,9 @@ public static function wysiwygText(?string $text, array $params = []): ?string
                             $site = Frontend::getSiteForDocument($element);
                             if ($site instanceof Site) {
                                 if (preg_match('~^' . preg_quote($site->getRootPath(), '~') . '~', $path)) {
-                                    $path = Tool::getRequestScheme() . '://' . $site->getMainDomain() . preg_replace('~^' . preg_quote($site->getRootPath(), '~') . '~', '', $path);
+                                    $path = preg_replace('~^' . preg_quote($site->getRootPath(), '~') . '$~', '', $path);
+                                    $path = preg_replace('~^' . preg_quote($site->getRootPath() . '/', '~') . '~', '/', $path);
```

Comment:
I don't think this works as intended. I assume it should be like this instead:

```suggestion
                                    $path = preg_replace('~^' . preg_quote($root, '~') . '([^/]|/)~', '\1', $path);
```